generate_compdb.py

```diff
@@ -62,15 +64,24 @@ def gen_compile_commands(out_dir):
         pool.join()
 
     print(file=sys.stderr)
-    print("Writing compile_commands.json...", file=sys.stderr)
+    print("Writing %s..." % (output, ), file=sys.stderr)
 
-    with open('compile_commands.json', 'w') as compdb_file:
+    with open(output, 'w') as compdb_file:
         json.dump(compdb, compdb_file, indent=1)
 
 
 def main():
     cmd_parser = argparse.ArgumentParser()
-    cmd_parser.add_argument('-O', '--out-dir', type=str, default=os.getcwd(), help="Build output directory")
+    default_out_dir = os.getcwd()
+    cmd_parser.add_argument(
+        '-O', '--out-dir',
+        type=str, default=default_out_dir,
+        help="Build output directory, default: %s (current directory)" % (default_out_dir, ))
+    default_output = os.path.abspath(os.path.join(os.path.dirname(__file__), 'compile_commands.json'))
+    cmd_parser.add_argument(
+        '-o', '--output',
+        type=str, default=default_output,
+        help="Path to resulting JSON file, default: %s" % (default_output, ))
```

Comment:
Please use [ArgumentDefaultsHelpFormatter](https://docs.python.org/3.5/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter) instead of formatting help strings by hand. It's available even in Python 2. You won't need `default_*` variables with it.